lib/helpers.js

```diff
@@ -220,3 +201,26 @@ function _addTypes({existingTypes, newTypes}) {
   }
   return [...types];
 }
+
+function _mergeProfileContent({profile, profileContent} = {}) {
+  let existingTypes;
+  if(profile.type.includes('Agent')) {
+    existingTypes = ['User', 'Agent'];
+  } else if(profile.type.includes('Profile')) {
+    existingTypes = ['User', 'Profile'];
+  }
+  return {
+    ...profileContent,
+    ...profile,
+    // special handle type to include required types
+    type: _addTypes({
+      existingTypes,
+      newTypes: profileContent.type
+    }),
+    // special handle zcaps to avoid overwrites
+    zcaps: {
+      ...profile.zcaps,
+      ...profileContent.zcaps
+    }
+  };
+}
```

Comment:
Hmm, if we're going to use this for profile agent content too, we should change the name and pass in the `existingTypes` as an option, rather than having the helper method guess what's going on. It's better to have the caller direct it on what to do even if we repeat ourselves with `existingTypes` in a couple of places. Alternatively, we could have `_mergeContent` that takes `existingTypes` and then two other helpers that pass in the appropriate `existingTypes`: `_mergeProfileContent` and `_mergeProfileAgentContent`.

This suggestion does the latter:

```suggestion
function _mergeContent({target, content, existingTypes} = {}) {
  return {
    ...content,
    ...target,
    // special handle type to include required types
    type: _addTypes({
      existingTypes,
      newTypes: content.type
    }),
    // special handle zcaps to avoid overwrites
    zcaps: {
      ...target.zcaps,
      ...content.zcaps
    }
  };
}

function _mergeProfileContent({profile, profileContent} = {}) {
  return _mergeContent({
    target: profile, content: profileContent,
    existingTypes: ['User', 'Profile']
  });
}

function _mergeProfileAgentContent({profileAgent, profileAgentContent} = {}) {
  return _mergeContent({
    target: profileAgent, content: profileAgentContent,
    existingTypes: ['User', 'Agent']
  });
}
```

Reply:
Thanks, I've added what you've suggested here [d139e38](https://github.com/digitalbazaar/bedrock-web-wallet/pull/28/commits/d139e3827ed15b263dbcc49e84241095c7c0fc24).